src/components/editor/DraftSpec.tsx

```diff
@@ -1,43 +1,60 @@
-import { updateDraftSpec } from 'api/draftSpecs';
+import { modifyDraftSpec } from 'api/draftSpecs';
 import MonacoEditor from 'components/editor/MonacoEditor';
 import {
     useEditorStore_currentCatalog,
-    useEditorStore_id,
+    useEditorStore_persistedDraftId,
     useEditorStore_setSpecs,
 } from 'components/editor/Store/hooks';
 import { useEntityType } from 'context/EntityContext';
 import useDraftSpecs, { DraftSpecQuery } from 'hooks/useDraftSpecs';
 import { useEffect, useState } from 'react';
-import { useResourceConfig_collections } from 'stores/ResourceConfig/hooks';
+import {
+    useResourceConfig_collections,
+    useResourceConfig_currentCollection,
+} from 'stores/ResourceConfig/hooks';
 
 export interface Props {
     disabled?: boolean;
+    localZustandScope?: boolean;
+    editorHeight?: number;
 }
 
-function DraftSpecEditor({ disabled }: Props) {
+function DraftSpecEditor({
+    disabled,
+    localZustandScope = false,
+    editorHeight,
+}: Props) {
     const entityType = useEntityType();
 
     // Draft Editor Store
-    const currentCatalog = useEditorStore_currentCatalog();
+    const currentCatalog = useEditorStore_currentCatalog({
+        localScope: localZustandScope,
+    });
 
-    const setSpecs = useEditorStore_setSpecs();
+    const setSpecs = useEditorStore_setSpecs({
+        localScope: localZustandScope,
+    });
 
-    const draftId = useEditorStore_id();
+    const draftId = useEditorStore_persistedDraftId();
 
     // Resource Config Store
+    const currentCollection = useResourceConfig_currentCollection();
     const collections = useResourceConfig_collections();
 
-    const { draftSpecs, mutate } = useDraftSpecs(draftId, null, entityType);
+    const { draftSpecs, mutate } = useDraftSpecs(
+        draftId,
+        null,
+        localZustandScope ? 'collection' : entityType
```

Comment:
This felt a bit _hacky_ to me. Passing the entity type as a prop seems like a better way to go here.

Reply:
Adjusted my approach to setting `currentCatalog` and `spec` local editor store state used in the `Editor` component.